packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py

```diff
@@ -202,18 +201,18 @@ def WatchMetadata(
         self.logger.exception('Exception calling the response handler. %s.', e)
 
   def GetMetadata(
-      self, metadata_key='', recursive=True, timeout=None, retry=True):
+      self, metadata_key='', recursive=True, timeout=None, retry_limit=None):
     """Retrieve the contents of metadata server for a metadata key.
 
     Args:
       metadata_key: string, the metadata key to watch for changes.
       recursive: bool, True if we should recursively watch for metadata changes.
       timeout: int, timeout in seconds for returning metadata output.
-      retry: bool, True if we should retry on failure.
+      retry_limit: int or None, limit for number of times to retry on failure.
```

Comment:
It could be useful to indicate here what happens when retry_limit is None (retry indefinitely ) and perhaps a note that setting retry_limit to 0 means "don't retry".

Reply:
Done.